src/tlo/methods/rti.py

```diff
@@ -4131,12 +4250,13 @@ def apply(self, person_id, squeeze_factor):
             if code[0] in df.loc[person_id, 'rt_injuries_for_open_fracture_treatment']:
                 df.loc[person_id, 'rt_injuries_for_open_fracture_treatment'].remove(code[0])
         else:
-            self.sim.modules['RTI'].schedule_hsi_event_for_tomorrow(self)
-            if pd.isnull(df.loc[person_id, 'rt_date_death_no_med']):
-                df.loc[person_id, 'rt_date_death_no_med'] = self.sim.date + DateOffset(days=7)
-            logger.debug(key='rti_general_message',
-                         data=f"Person {person_id}'s has {open_fracture_counts} open fractures without treatment",
-                         )
+            if self._number_of_times_this_event_has_run < self._maximum_number_times_event_should_run:
+                self.sim.modules['RTI'].schedule_hsi_event_for_tomorrow(self)
+                if pd.isnull(df.loc[person_id, 'rt_date_death_no_med']):
+                    df.loc[person_id, 'rt_date_death_no_med'] = self.sim.date + DateOffset(days=7)
+                logger.debug(key='rti_general_message',
+                             data=f"Person {person_id}'s has {open_fracture_counts} open fractures without treatment",
+                             )
```

Comment:
as per the comment above, I think this need to be outside of the new clause we're introducing about the rescheduling (or not) the HSI event.